### PR TITLE
Match energy binning per decade to pyirf's

### PR DESCRIPTION
**Pull request by JBernete**
**Description**

This pull request has the purpose of matching what pyirf does when making an energy binning per decade, i.e. to make sure that when using the same energy limits and number of bins per decade, the resulting bins are exactly the same in both pyirf and gammapy.
Before these changes, the value of energy_max was always included as the last edge, even when it didn't match the bins-per-decade requirement, so the edges were shifted and the actual number of bins per decade was a fraction higher. Now, energy_max is not forced to be an edge and the last edge is recalculated.

**Dear reviewer**

This PR is ready for review.

### CODE REVIEW

**Comment by QRemy** [comment]
> Maybe another factory method with this specific behavior can be introduced if this is necessary.

or another option `strict_bounds`=True that produce the current behaviour by default  so it is backward compatible.

**Comment by JBernete** [comment]
> > Maybe another factory method with this specific behavior can be introduced if this is necessary.
> 
> or another option `strict_bounds=True` that produce the current behaviour by default so it is backward compatible.

I like this idea. I added the option. Please review it again.

File: gammapy/maps/axes.py [hunk 1]
```diff
@@ -564,6 +565,10 @@ def from_energy_bounds(
             Whether `nbin` is given per decade.
         name : str
             Name of the energy axis, either 'energy' or 'energy_true'
+        strict_bounds : bool
+            Whether to strictly end the binning at the max energy when
+            `per_decade=True`. If True, the number of bins per decade
+            might be slightly adjusted.
```
Comment (AtreyeeS) [inline gammapy/maps/axes.py:571]: Maybe document explicitly the `False` option as well

**Review by registerrier — COMMENTED** [review]
Thanks @JBernete . This is nice.

I think you need to run black on your commit. You could install the pre-commit hooks following: 
https://docs.gammapy.org/1.1/development/intro.html#get-set-up

Could you just add a simple test to make sure the option is correctly taken into account?

**Review by registerrier — APPROVED** [review]
Thanks @JBernete . This looks good. I have no further comments.

**Pull request merged**